Framework/DataHandling/src/LoadBBY.cpp

```diff
@@ -498,21 +515,40 @@ void LoadBBY::loadInstrumentParameters(
         // comma separated details
         std::vector<std::string> details;
         boost::split(details, line, boost::is_any_of(","));
+        if (details.size() < 3) {
+          g_log.warning() << "Invalid format for BILBY parameter " << x.first
+                          << std::endl;
+          continue;
+        }
         auto hdfTag = boost::algorithm::trim_copy(details[0]);
         try {
-          // extract the parameter and add it to the parameter dictionary
-          if (!hdfTag.empty() && loadNXDataSet(entry, hdfTag, tmpFloat)) {
-            auto factor = std::stod(details[1]);
-            logParams[logTag] = factor * tmpFloat;
-          } else if (details.size() > 2) {
-            logParams[logTag] = std::stod(details[2]);
+          // extract the parameter and add it to the parameter dictionary,
+          // check the default for value numeric and string
+          auto updateOk = false;
+          if (!hdfTag.empty()) {
+            if (isNumeric(details[2])) {
+              if (loadNXDataSet(entry, hdfTag, tmpFloat)) {
+                auto factor = std::stod(details[1]);
+                logParams[logTag] = factor * tmpFloat;
+                updateOk = true;
+              }
+            } else {
```

Comment:
Could use `else if` here to reduce the nesting slightly

Reply:
Dropped one level by using else if {}. Thanks.